GoEddie/SQLCover
# dotnet core finished, SQLCover worked in linux, ready to publish as dotnet tool

Fixes #53 

## Details
* please add hactober label #59 
* had issues w/ database password in yml, so sometimes I used env variable and often I had to hard coded it in. I tried so many different combinations, I think it's inserting a special character which is causing my password authentication issues, pipeline fully passes with current solution
* package is ready to package as a **dotnet too**l see https://www.meziantou.net/how-to-publish-a-dotnet-global-tool-with-dotnet-core-2-1.htm
  * `cd src\SQLCover\SqlCoverCore`
  * `dotnet pack` using dotnet core 3
  * make sure you have your nuget organization registered and publish following website

## Changes proposed in this pull request:
 - Moved to a **shared library** that still works w/ powershell .net461 build and dotnet core ~~**copied** SqlCover project into SqlCoverCore~~
 - built a commandline **Program.cs** to parse commands to run SqlCover
 - most commands supported and pretty flushed out
 - very descriptive on commands needed
 - some label cleanup in pipeline
 - github actions demonstrates SQLCoverCore working on linux and using dotnet core
 - can also debug project inside of visual studio code.
 - changes in future to SQLCoverCore will be much easier b/c pipeline validates it.

## How to test this code:
 - see **github actions and dotnetcore.yml**, Export-Html, Export-OpenXml, and reportgenerator all working
   * also see github actions build artifacts for saved reports
 - 
 - 

Has been tested on (remove any that don't apply):
 - SQL Server 2019

## next steps
* besides publish tool
* add CoreTests
  * run c# tests against db in linux
* update readme
* not all SQLCover commands exposed, may need to expand or alter the command line arguments
